Socket connections may not actually fill byte[]

datariot opened the pull request:
Because of network latency, packet size, etc. the byte[] may not be filled on a single read() call for larger message sizes.

This while loop will ensure that the full message body is read.

datariot committed:
Socket connections may not actually fill byte[]

Because of network latency, packet size, etc. the byte[] may not be filled on a single read() call for larger message sizes.

This while loop will ensure that the full message body is read.

chrisguitarguy commented:
Thanks!

I made a few changes -- what you sent didn't compile at first -- just a wayward variable name. Also added a test (e88ad092f5c3c86ca0475f2c501aff14d655b3b8) with a large job body.

datariot commented:
Awesome! Sorry I pasted directly into GitHub from a fork that's been
modified for internal use. Cheers!

On Feb 1, 2014, at 9:13 AM, Christopher Davis notifications@github.com
wrote:

Thanks!

I made a few changes -- what you sent didn't compile at first -- just a
wayward variable name. Also added a test
(e88ad09https://github.com/chrisguitarguy/Beanstalkc/commit/e88ad092f5c3c86ca0475f2c501aff14d655b3b8)
with a large job body.

## 

Reply to this email directly or view it on
GitHubhttps://github.com/chrisguitarguy/Beanstalkc/pull/2#issuecomment-33875519
.

chrisguitarguy commented:
> Sorry I pasted directly into GitHub from a fork that's been modified for internal use.

No worries. I'm glad someone is using it!
